File: src/main/java/sharev/gathering/repository/IntroduceTemplateRepository.java
package sharev.gathering.repository;

import java.util.Optional;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;
import sharev.gathering.entity.IntroduceTemplate;

public interface IntroduceTemplateRepository extends JpaRepository<IntroduceTemplate, Long> {

    Optional<IntroduceTemplate> findByGatheringIdAndVersion(UUID gatheringId, int version);

    Optional<IntroduceTemplate> findTopByGatheringIdOrderByVersionDesc(UUID gatheringId);
}

File: src/main/kotlin/sharev/gathering/adapter/outbound/jpa/repository/IntroduceTemplateRepository.kt
package sharev.gathering.adapter.outbound.jpa.repository

import org.springframework.data.jpa.repository.JpaRepository
import sharev.gathering.adapter.outbound.jpa.entity.IntroduceTemplateJpaEntity
import java.util.*

interface IntroduceTemplateRepository : JpaRepository<IntroduceTemplateJpaEntity, Long> {
    fun findByGatheringIdAndVersion(gatheringId: UUID, version: Int): Optional<IntroduceTemplateJpaEntity>

    fun findTopByGatheringIdOrderByVersionDesc(gatheringId: UUID): Optional<IntroduceTemplateJpaEntity>
}
